File: internal/wikidata/sparql_test.go
package wikidata

import (
	"strings"
	"testing"
)

func TestShortenWikidataEntityURI(t *testing.T) {
	got := shortenWikidataEntityURI("http://www.wikidata.org/entity/Q42")
	if got != "Q42" {
		t.Fatalf("expected shortened URI to be Q42, got %q", got)
	}

	untouched := shortenWikidataEntityURI("https://example.com")
	if untouched != "https://example.com" {
		t.Fatalf("expected non-wikidata URI to stay unchanged, got %q", untouched)
	}
}

func TestToSemicolonCSV(t *testing.T) {
	csvText, err := toSemicolonCSV(
		[]string{"human", "humanLabel"},
		[]map[string]string{
			{"human": "Q42", "humanLabel": "Douglas Adams"},
			{"human": "Q23", "humanLabel": "George Washington"},
		},
	)
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}

	expectedLines := []string{
		";human;humanLabel",
		"0;Q42;Douglas Adams",
		"1;Q23;George Washington",
	}
	for _, line := range expectedLines {
		if !strings.Contains(csvText, line) {
			t.Fatalf("expected CSV output to contain %q, got:\n%s", line, csvText)
		}
	}
}

File: test/sparql.test.ts
import assert from "node:assert/strict";
import test from "node:test";

import { shortenWikidataEntityURI, toSemicolonCSV } from "../src/client.js";

test("shortenWikidataEntityURI", () => {
  const got = shortenWikidataEntityURI("http://www.wikidata.org/entity/Q42");
  assert.equal(got, "Q42");

  const untouched = shortenWikidataEntityURI("https://example.com");
  assert.equal(untouched, "https://example.com");
});

test("toSemicolonCSV", () => {
  const csvText = toSemicolonCSV(
    ["human", "humanLabel"],
    [
      { human: "Q42", humanLabel: "Douglas Adams" },
      { human: "Q23", humanLabel: "George Washington" },
    ],
  );

  const expectedLines = [
    ";human;humanLabel",
    "0;Q42;Douglas Adams",
    "1;Q23;George Washington",
  ];

  for (const line of expectedLines) {
    assert.match(csvText, new RegExp(line.replace(/[.*+?^${}()|[\]\\]/g, "\\$&")));
  }
});
